e2e/nomostest/config_sync.go

```diff
@@ -710,8 +710,47 @@ func setupDelegatedControl(nt *NT) {
 	}
 }
 
-// RootSyncObjectV1Alpha1Git returns the default RootSync object.
-func RootSyncObjectV1Alpha1Git(name, repoURL string, sourceFormat configsync.SourceFormat) *v1alpha1.RootSync {
+// SetRSyncTestDefaults populates defaults for test RSyncs.
+func SetRSyncTestDefaults(nt *NT, obj client.Object) {
+	isRepoSync := false
+	// Set type-specific defaults
+	switch rs := obj.(type) {
+	case *v1alpha1.RootSync:
+		if nt.DefaultReconcileTimeout != nil && (rs.Spec.Override == nil || rs.Spec.Override.ReconcileTimeout != nil) {
```

Comment:
Did you intend to change the behavior with this refactor? It doesn't look like this conditional is equivalent the the previous one.

e.g. if `nt.DefaultReconcileTimeout is nil` and `rs.Spec.Override is not nil`

Reply:
I did. The old behavior didn't look right. So I updated the behavior to match the name. 

It shouldn't matter though. This func is only ever used on newly constructed objects, so the other case is never being executed.

If someone decides to use it in the future, it will now act as a default, rather than an override.

Reply:
Oops. Typo on that last bit. Fixed now.

It should set the timeout as a default now only if either Override is nil or Override.ReconcileTimeout is nil.